src/conn_poll.c

```diff
@@ -151,20 +155,22 @@ int conn_poll_prepare(conn_registry_t *registry, pfds_record_t *pfds, timestamp_
 #endif
 	interrupt_pfd->events = POLLIN;
 
-	for (nfds_t i = 1; i < pfds->size; ++i) {
+	for (nfds_t i = 1; i < pfds->size; i += 2) {
 		struct pollfd *pfd = pfds->pfds + i;
-		juice_agent_t *agent = registry->agents[i - 1];
+		struct pollfd *pfd_tcp = pfds->pfds + i + 1;
```

Comment:
This goes out of bounds since `pfds` has size `1 + registry->agents_size` (for one agent, `pfd_tcp` points to index 2 while the array has size 2).

Reply:
Additionally, most people won't use ICE TCP so this approach is quite wasteful memory-wise. If you plan to implement TCP passive later, you'd need 3 entries per agent so it would be even worse. 

I think you should instead iterate on agents while keeping an index for `pfds` in `conn_poll_prepare` and `conn_poll_recv`, and add a TCP socket only if the agent has one. This would greatly simplify socket indexing and prevent useless entries.

Reply:
Fixed! I instead iterate off of Agents now.